metering-billing/lambdas/src/main/java/com/amazon/aws/partners/saasfactory/metering/onboarding/BillingIntegration.java

```diff
@@ -271,58 +272,94 @@ private void provisionTenantInStripe(String tenantId, String planId) throws Stri
+                                //.setEmail("bogus@example.com")
+                                .setPaymentMethod("pm_card_visa")
+                                .setInvoiceSettings(
+                                        CustomerCreateParams.InvoiceSettings.builder()
+                                                .setDefaultPaymentMethod("pm_card_visa")
+                                                .build())
+                                .setDescription(customerName + " from AWS SaaS Boost")
+                                .build();
+                customer = Customer.create(params);
+                //create the subscription
+                createStripeSubscription(customer, planEnum, tenantId,false);
+                LOGGER.info("provisionTenantInStripe: Customer created with id: " + customer.getId());
+            } catch (StripeException e) {
+                LOGGER.error("provisionTenantInStripe: Error creating Customer for Tenant: " + tenantId);
+                LOGGER.error(Utils.getFullStackTrace(e));
+                throw e;
+            }
+        } else {
+            LOGGER.info("provisionTenantInStripe: Customer: {} create skipped, already exists with id: {} ", customerName, customer.getId());
+            //If subscription plan has changed then cancel the old subscriptions and create a new.
+            //Ideally, the subscription information for the tenant would be stored in a DDB table so we do not have to hit Stripe to find if it has changed
+
+            //Compare plan with what is on subscription item in Stripe to determine if plan has changed
+
+            //get subscriptions for customer
+            try {
+                final SubscriptionListParams subscriptionListParams = SubscriptionListParams.builder().setCustomer(customer.getId()).build();
+                final SubscriptionCollection collection = Subscription.list(subscriptionListParams);
+                for (final Subscription subscription : collection.getData()) {
+                    LOGGER.info("provisionTenantInStripe: check Stripe subscription items for matching plan");
+                    for (SubscriptionItem si : subscription.getItems().getData()) {
+                        if (si.getPrice().getProduct().equalsIgnoreCase(planId)) {
+                            LOGGER.info("Subscription Plan has not changed for the tenant, nothing to do");
+                            return;
+                        }
+                    }
+                    //NOTE: Do queued metered events need to sent to Stripe before canceling the subscription?
+
+                    //attempt to cancel the subscription
+                    LOGGER.info("provisionTenantInStripe: No match in Stripe subscription items for plan {}", planId);
+                    LOGGER.info("provisionTenantInStripe: Attempt to Cancel Subscription: {} ", subscription.getId());
+                    Subscription canceledSubscription = subscription.cancel();
```

Comment:
Why not [update the subscription](https://stripe.com/docs/api/subscriptions/update) instead of cancelling and recreating?

Reply:
Both great questions. Sounds like when we pick up this PR we'll need to do some experimentation with Stripe's API to see how it acts in different situations.

Reply:
Upon further research, Stripe does allow updating the subscription with a new plan and it will do the Proration. 
Take a look at https://stripe.com/docs/billing/subscriptions/change and https://stripe.com/docs/billing/subscriptions/prorations. May be we need to look at this approach instead of the cancel and create new subscription.